docs/design/features/Apple-Silicon-UX.md

```diff
@@ -0,0 +1,107 @@
+# Apple Silicon User Experience
+
+Apple has [announced plans to transition its Mac hardware line]( https://www.apple.com/newsroom/2020/06/apple-announces-mac-transition-to-apple-silicon/) to a new Arm64-based chip that they refer to as “Apple Silicon”.
+
+The transition has a few impacts on the .NET user experience.
+
+## Mixed Architecture Side By Side Installs
+
+Support for a new processor architecture is a significant undertaking.  It is too large an undertaking to consider on our stable release branches.  New architecture development is occurring in .NET 6.
+
+We do not plan to back-port Apple Silicon support to the current releases .NET 2.1, .NET 3.1, and .NET 5.0.
+
+As part of the their transition plan, Apple created the Rosetta 2 x86_64 emulator to run x64 code on an Apple Silicon processor. This means that .NET 2.1 LTS, .NET 3.1 LTS and .NET 5 mostly just work on Apple Silicon.
+
+However the `dotnet` command line architecture makes it difficult to support side-by-side mixed architecture installs. The dotnet tool is responsible for stating the `hostfxr` library to resolve the runtime version to use. Until the runtime version is selected it is not clear which architecture the runtime will require.
+
+For instance, the `dotnet --info` command must resolve the currently selected SDK, by finding the `global.json` or the most recent SDK. Then it must find the best runtime as requested by that SDK. Once that is determined the runtime required architecture could be known.
+
+### Other Observations
+
+- This mixed architecture problem is transient in nature. Essentially it means it will only exist for the support lifetime of the previous releases (2.1, 3.1, 5.0).  We want to avoid making major architectural changes to support this temporary design problem.
+
+- We want to keep the process parent-child relationships as consistent as possible.
+
+- We want minimal changes to the earlier releases.
+
+### Proposed Side-By-Side Design
+
+For Apple Silicon the proposed plan is to:
+
+- On Apple Silicon .NET runtime versions less than 6.0 will always require `x86_64` architecture. All other runtime versions will require `arm64` architecture on Apple Silicon.
+
+- Make the .NET 6 `dotnet` & `libhostfxr.dylib` (`hostfxr`) universal binaries. This means both native and emulated processes can perform runtime resolution to identify the correct runtime to use.
+
+- The `hostfxr` will be modified to enforce the architecture version  policy above. It will return an architecture mismatch error when the runtime (by version policy above) require a different architecture than the currently running process.
```

Comment:
Would that be enough for good user experience? What will happen when someone installs and runs for example dotnet tool on Apple Silicon which was built for x86_64?

Reply:
That should just work for a .NET 5/.NET 3.1 tool.  For a .NET 6.0 tool built for x64, it really depends.  My original position was to disable it.  @jkotas is suggesting making it work if `DOTNET_ROOT` is set.  So it is still TBD.